src/rpg_database/character_sheet/FocusesSet.java

```diff
@@ -0,0 +1,171 @@
+		return focusesSet.size();
+	}
+
+	@Override
+	public boolean isEmpty() {
+		return focusesSet.isEmpty();
+	}
+
+	@Override
+	public boolean contains(Object o) {
+		return focusesSet.contains(o);
+	}
+
+	@Override
+	public Iterator<Focus> iterator() {
+		return focusesSet.iterator();
+	}
+
+	@Override
+	public Object[] toArray() {
+		return focusesSet.toArray();
+	}
+
+	@Override
+	public <T> T[] toArray(T[] a) {
+		return focusesSet.toArray(a);
+	}
+
+	/**
+	 * Does some thing in old style.
+	 *
+	 * @deprecated use {@link #new()} instead.
+	 */
+	@Deprecated
+	@Override
+	public boolean add(Focus focus) {
+		return focusesSet.add(focus);
+	}
+
+	public boolean add(Focuses focus) {
+		Focus focusInDb = getFocus(focus, focusesSet);
+		if (focusInDb == null) {
+			Focus tempFocus = new Focus(focus);
+			focusesSet.add(tempFocus);
+			return tempFocus.isFocusImproved();
+		} else {
+			if (focusInDb.isFocusImproved()) {
+				throw new InvalidFocusesSetException(focus.name() + " is already improved!");
+			}
+			focusInDb.makeFocusImproved();
+			return focusInDb.isFocusImproved();
+		}
+	}
+
+	// TODO comment dekorator needed
+	public boolean addAllFocuses(Collection<? extends Focuses> focusesCollection) {
+		return true;
+	}
+
+	@Override
+	public boolean remove(Object o) {
+		return focusesSet.remove(o);
+	}
+
+	@Override
+	public boolean containsAll(Collection<?> focusesCollection) {
+		return focusesSet.containsAll(focusesCollection);
+	}
+
+	/**
+	 * Does some thing in old style.
+	 *
+	 * @deprecated use {@link #new()} instead.
+	 */
+	@Deprecated
+	@Override
+	public boolean addAll(Collection<? extends Focus> focusesCollection) {
+		return focusesSet.addAll(focusesCollection);
+	}
+
+	@Override
+	public boolean retainAll(Collection<?> c) {
+		return focusesSet.retainAll(c);
+	}
+
+	@Override
+	public boolean removeAll(Collection<?> c) {
+		return focusesSet.removeAll(c);
+	}
+
+	@Override
+	public void clear() {
+		focusesSet.clear();
+	}
+
+	@Override
+	public boolean equals(Object obj) {
+		return focusesSet.equals(obj);
+	}
+
+	@Override
+	public String toString() {
+		ArrayList<String> focusesNames = new ArrayList<>();
+		for (Focus focuse : focusesSet) {
+			focusesNames.add(focuse.toString());
+		}
+		return String.join(", ", focusesNames);
+	}
+
+	private HashSet<Focus> findDuplicates(Focuses... focuses) {
+		final HashSet<Focus> setToReturn = new HashSet<Focus>();
+		final HashSet<Focus> setTemp = new HashSet<Focus>();
+
+		for (Focuses focus : focuses) {
+			Focus tempFocus = new Focus(focus);
+			if (!setTemp.add(tempFocus)) {
+				Focus focusEnum = getFocus(focus, setToReturn);
+				focusEnum.makeFocusImproved();
+			} else {
+				setToReturn.add(tempFocus);
+			}
+		}
+		return setToReturn;
+	}
+
+	private Focus getFocus(Focuses focus, HashSet<Focus> focusesList) {
+		for (Focus containedFocus : focusesList) {
+			if (containedFocus.getFocus().equals(focus))
+				return containedFocus;
+		}
+		return null;

```

Comment:
Ez vezethet hosszútávon nullpointerexception-höz. Én inkább exception handlinggel oldottam volna meg, mert ahol használod ott el tudod kapni és tudod kezelni, cserébe ha olyan függvény használja ami nem várt volna tőle null-t az dob egy célirányozott üzenetű exception-t sima NullPointerException helyett vagy kénytelen kezelni, így védve van a program. Just my two cents, de rád bízom hogy van -e igényed ezt átírni.

Reply:
Igen, egyetértek és átírtam. .